packages/editor-ui/src/components/NodeCreator/helpers.ts

```diff
@@ -0,0 +1,179 @@
+import { CORE_NODES_CATEGORY, CUSTOM_NODES_CATEGORY, SUBCATEGORY_DESCRIPTIONS, UNCATEGORIZED_CATEGORY, UNCATEGORIZED_SUBCATEGORY, HIDDEN_NODES  } from '@/constants';
+import { INodeCreateElement, ICategoriesWithNodes, INodeItemProps } from '@/Interface';
+import { INodeTypeDescription } from 'n8n-workflow';
+
+
+export const getCategoriesWithNodes = (nodeTypes: INodeTypeDescription[]): ICategoriesWithNodes => {
+	const categorized = nodeTypes.reduce(
```

Comment:
no need to assign it to a variable first
```suggestion
	return nodeTypes.reduce(
```